add ADR 0002: code platform (#28)
MinBZK/ai-validation: docs/ADRs/0002-code-platform.md

---
    title: 0002 Code Platform
    adr:
        author: Berry den Hartog
        created: 28-Feb-2024

>         status:  proposed

Reviewer: Maybe we should set this to `accepted`, pending this review, of course.

---

## Context

In the landscape of software development, the choice of coding platform significantly impacts developer productivity,
collaboration, and code quality. it's crucial to evaluate and select a coding platform that aligns with our development
needs and fosters efficient workflows.

## Assumptions

The following assumptions are made:

> - our work should be visible to external teams to promote transparency and facilitate collaboration.

Author: Solved

> - The coding platform should be easily available for developers

Author: Solved

> - the coding platform should offert collaboration tools between developers and the community

Reviewer 2: ```suggestion
- The coding platform should offert collaboration tools between developers and the community.
```

Author: Solved

> - the coding platform should offert security and dependecy management tools

Author: Solved

> - the coding platform should have low costs

Reviewer 2: A good Cost/Efficiency trade-off, as we are currently also debating where we would opt for the paid version of Github? Or would that be a new ADR?

Reviewer 3: allthough I thinks this is a good point, it is open for debate 'what is low' and what is good 'cost/efficiency'.. so maybe this should be more politically rephrased (allthough it becomes a bit more meaningless as well):
- the pricing model should be suitable for our budget and needs

Author: i checked the licences. currenty it does give low benefits for a github licences

## Decision

> After careful consideration and evaluation of various options like GitHub, gitlab and bitbucket, we propose adopting

Reviewer 2: ```suggestion
After careful consideration and evaluation of various options like GitHub, Gitlab, and Bitbucket, we propose adopting
```

> After careful consideration and evaluation of various options like github, gitlab and bitbucket, we propose adopting
> Github as our primary coding platform. The decision is based on the following factors:

Reviewer: ```suggestion
After careful consideration and evaluation of various options like GitHub, GitLab and Bitbucket, we propose adopting
GitHub as our primary coding platform. The decision is based on the following factors:
```

> **Costs**: there are currenlty no costs associate in using GitHub for our usecases

Reviewer 2: ```suggestion
**Costs**: There are currenlty no costs associate in using GitHub for our usecases.
```

> **Features and Functionality**: Github offers a comprehensive set of features essential for modern software development

Reviewer: ```suggestion
**Features and Functionality**: GitHub offers a comprehensive set of features essential for modern software development
```

and collaboration with external teams, including version control, code review, issue tracking, continuous integration,
and deployment automation.

**Security**: Github offers a complete set of security features essential to secure development like dependency management
and security scanning.

> **Community and Ecosystem**: Github boasts a vibrant community and ecosystem, facilitating knowledge sharing, collaboration,

Reviewer: ```suggestion
**Community and Ecosystem**: GitHub boasts a vibrant community and ecosystem, facilitating knowledge sharing, collaboration,
```

> and access to third-party tools and services that can enhance our development workflows. Within our organization we have

Reviewer 2: ```suggestion
and access to third-party tools, and services that can enhance our development workflows. Within our organization we have
```

easy access to the team managing the GitHub organization.

> **Usability and User Experience**: A user-friendly interface and intuitive workflows are essential for maximizing developer

Reviewer 2: ```suggestion
**Usability and User Experience**: An user-friendly interface and intuitive workflows are essential for maximizing developer
```

Reviewer: it's "a user", not "an user"

productivity and minimizing onboarding time. Github offers a streamlined user experience and customizable workflows that
align with our team's preferences and practices.

## Risks

> Currenlty the organization of MinBZK on github does not have allot of `people` indicating that our team is an early

Reviewer: ```suggestion
Currently the organization of MinBZK on GitHub does not have allot of `people` indicating that our team is an early
```

> Currently the organization of MinBZK on GitHub does not have allot of `people` indicating that our team is an early

Reviewer 2: ```suggestion
Currently the organization of MinBZK on GitHub does not have a lot of `people` indicating that our team is an early
```

adapter of the platform within the organization. This might impact our features due to cost constrains.

## Consequences

> If we choose another tool in the future we need to migrate our codebase, and maybe need to rewrite some specific GitHub

Reviewer 2: ```suggestion
If we choose another tool in the future we need to migrate our codebase, and potentially need to rewrite some specific GitHub
```

> If we chooice another tool in the future we need to migrate our codebase, and maybe need to rewrite some specific Github

Reviewer: ```suggestion
If we choose another tool in the future we need to migrate our codebase, and maybe need to rewrite some specific GitHub
```

> features that can be used.

Reviewer 2: ```suggestion
features that cannot be used in another tool.
```

## More Information

Alternatives considered:

- Bitbucket

> - GitLab

Reviewer 2: ```suggestion
- GitLab
- [Forgejo](https://forgejo.org/)
```

> - bitbucket
> - gitlab

Reviewer: ```suggestion
- Bitbucket
- GitLab
```
